collector/aws/resources/apigateway_test.go

```diff
@@ -0,0 +1,255 @@
+package resources
+
+import (
+	"errors"
+	awsTestutils "finala/collector/aws/testutils"
+	"finala/collector/config"
+	"finala/collector/testutils"
+	collectorTestutils "finala/collector/testutils"
+	"reflect"
+	"testing"
+	"time"
+
+	awsClient "github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/apigateway"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+)
+
+var mockApiGatways = apigateway.GetRestApisOutput{
+	Items: []*apigateway.RestApi{
+		{
+			Id:          awsClient.String("foo-id"),
+			Name:        awsClient.String("foo"),
+			CreatedDate: testutils.TimePointer(time.Now()),
+			Tags: map[string]*string{
+				"tag-foo-1": awsClient.String("tag-1"),
+				"tag-foo-2": awsClient.String("tag-2"),
+			}},
+		{
+			Id:          awsClient.String("bar-id"),
+			Name:        awsClient.String("bat"),
+			CreatedDate: testutils.TimePointer(time.Now()),
+			Tags: map[string]*string{
+				"tag-bar-1": awsClient.String("tag-1"),
+				"tag-bar-2": awsClient.String("tag-2"),
+			}},
+	},
+}
+
+type mockAPIGatewayCLient struct {
+	response    apigateway.GetRestApisOutput
+	errResponse error
+}
+
+func (mg *mockAPIGatewayCLient) GetRestApis(input *apigateway.GetRestApisInput) (*apigateway.GetRestApisOutput, error) {
+
+	return &mg.response, mg.errResponse
+}
+
+func TestNewAPIGatewayManager(t *testing.T) {
+
+	collector := collectorTestutils.NewMockCollector()
+	detector := awsTestutils.AWSManager(collector, nil, nil, "us-east-1")
+
+	t.Run("valid", func(t *testing.T) {
+
+		mockClient := mockAPIGatewayCLient{}
+
+		apigateway, err := NewAPIGatewayManager(detector, &mockClient)
+		if err != nil {
+			t.Fatalf("unexpected error happened, got %v expected nil", err)
+		}
+		if reflect.TypeOf(apigateway) != reflect.TypeOf(&APIGatewayManager{}) {
+			t.Fatalf("unexpected apigateway manager instance, got %v expected %v", reflect.TypeOf(apigateway), reflect.TypeOf(&APIGatewayManager{}))
+		}
+
+	})
+	t.Run("error", func(t *testing.T) {
+
+		mockClient := MockEmptyClient{}
```

Comment:
Why are you using MockEmptyClient{} of docbdb_test ?

Reply:
done